File: 006/rock_paper_scissors.js
var prompt = require('prompt');

//This sets each players score to 0
var p1_score = 0;
var p2_score = 0;

//This calls the begingame function to begin the game
beginGame();

//This function checks for a game winner and starts the next round if non found
function beginGame(){
	

	if ((p1_score > 0) || (p2_score > 0)) {
		console.log()
		console.log("The score is Player 1: " + p1_score + " and Player 2: " + p2_score);
		
	}

	if (p1_score == 3){

 		console.log("Game Over! Player 1 wins");

	} else if (p2_score == 3){

		console.log("Game Over! Player 2 wins");

	} else {
		console.log();
		console.log("Player 1, please enter a weapon!");
		player1();

	}
}


//This functin gets the result of player 1
function player1(){

	prompt.get(['player1'], function (err, result) {

	 	//Makes sure the entered ammounts are rock, paper or scissors
		var choices = ['rock','paper', 'scissors'];
		if(illegal_words(result.player1)){
			console.log();
		    console.log("Player 1, please enter a weapon!");
		    
			player2(result.player1);

		} else {
			console.log();
			console.log("It seems you can't spell. Please type either rock, paper, or scissors!");
			beginGame();
		}

	}
	)
	;
}

//This function gets the result of player 2
function player2(results_player1){

	prompt.get(['player2'], function (err, result) {
	     

	 	//Makes sure the entered ammounts are rock, paper or scissors
		var choices = ['rock','paper', 'scissors'];
		if(illegal_words(result.player2)){
			compare(results_player1, result.player2);
		} else {
			console.log();
			console.log("It seems you can't spell. Please type either rock, paper, or scissors!");
			beginGame();
		}

	}
	)
	;
}


//This function checks who won
function compare(option1, option2) {

	//Win logic. 
	if (option1 == option2) {
		console.log("It's a Tie! Try again!")

	} else if (option1 == "rock" && option2 == "paper") {
		console.log("Player 2 Wins!")
		p2_score ++;

	} else if (option1 == "scissors" && option2 =="rock") {
		console.log("Player 2 Wins!")
		p2_score ++;

	} else if (option1 == "paper" && option2 =="scissors") {
		console.log("Player 2 Wins!")
		p2_score ++;

	} else {
		console.log("Player 1 Wins!")
		p1_score ++;
	}

	beginGame()
}   

//This function checks if a rock, paper or scissors was entered
function illegal_words(a){
	var choices = ['rock','paper', 'scissors'];
	for(i=0; i<choices.length; i++){
      if(choices[i] == a)
      return true;
	}
	return false;
}

File: 008/rock_paper_scissors.rb
require "pry"

$p1_total_score = 0
$p2_total_score = 0


class NewGame

	# checks to see if there is a winner. If not, coninues game.
	def declare_winner(p1_total_score, p2_total_score)
		if $p1_total_score >= 3
			puts
			puts "Player 1 is the champian!"
			puts "Player 1 won #{$p1_total_score} to #{$p2_total_score}"
		elsif $p2_total_score >= 3
			puts
			puts "Player 2 is the champian!"
			puts "Player 2 won #{$p2_total_score} to #{$p1_total_score}"
		else
			annouce_score()
			player1()
		end
	end

	# Prompts Player 1 to pick a weapon and saves the results to a variable
	def player1
		print "Player 1, pick your weapon! "
		p1_answer = gets.chomp
		p1_answer = p1_answer.downcase
		if quality_control(p1_answer)
			player2(p1_answer)
		else
			print "'#{p1_answer}'' is not a real weapon! "
			player1()
		end
	end

	# Prompts Player 2 to pick a weapon and saves the results to a variable
	def player2(p1_answer)
		print "Player 2, pick your weapon! "
		p2_answer = gets.chomp
		p2_answer = p2_answer.downcase
		if quality_control(p2_answer)
			round_win(p1_answer, p2_answer)
		else 
			print "'#{p2_answer}'' is not a real weapon! "
			player2(p1_answer)
		end
	end

	# Makes sure the player enter a valid response
	def quality_control(answer)
		if answer == "cheat code"
			$p1_total_score = 1000000 - 1 
			return true
		elsif answer == "rock" or answer == "scissors" or answer == "paper"
			return true
		else
			return false
		end	
	end

	# Winner logic. Determines who the winner is.
	def round_win(p1_answer, p2_answer)
		if p1_answer == p2_answer
			puts "It's a tie! Try again!"
			player1()
		elsif p1_answer == "rock" and p2_answer == "paper"
			puts "Player 2 wins this round!"
			tally_score("p2")
		elsif p1_answer == "scissors" and p2_answer == "rock"
			puts "Player 2 wins this round!"
			tally_score("p2")
		elsif p1_answer == "paper" and p2_answer == "scissors"
			puts "Player 2 wins this round!"
			tally_score("p2")
		else 
			puts "Player 1 wins this round!"
			tally_score("p1")
		end
	end

	# Depending who won the round, will increment their score
	def tally_score(winner)
		if winner == "p2"
			$p2_total_score += 1
		else
			$p1_total_score += 1   
		end
			declare_winner($p1_total_score, $p2_total_score)
	end 
	# Prints the current score
	def annouce_score
		if $p1_total_score < $p2_total_score
			puts
			puts "Player 2 is in the lead with #{$p2_total_score} to #{$p1_total_score}!"
			puts
		elsif $p1_total_score > $p2_total_score
			puts
			puts "Player 1 is in the lead with #{$p1_total_score} to #{$p2_total_score}!"
			puts
		else
			puts
			puts "We have a tie! the score is #{$p1_total_score} to #{$p2_total_score}"
			puts
		end
	end
end

newGame = NewGame.new
newGame.player1()





